File: server/router/octopus/interval_task.go
package octopus

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type IntervalTaskRouter struct{}

func (s *IntervalTaskRouter) InitIntervalTaskRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	intervalTaskRouter := Router.Group("intervalTask").Use(middleware.OperationRecord())
	var intervalTaskApi = v1.ApiGroupApp.OctopusApiGroup.IntervalTaskApi

	{
		intervalTaskRouter.POST("createIntervalTask", intervalTaskApi.CreateIntervalTask)   // 新建任务
		intervalTaskRouter.DELETE("deleteIntervalTask", intervalTaskApi.DeleteIntervalTask) // 删除任务
		intervalTaskRouter.GET("stopIntervalTask", intervalTaskApi.StopIntervalTask)        // 停止单个任务
	}

}

File: web/src/api/octopus/intervalTask.js
import service from '@/utils/request'

export const createIntervalTask = (data) => {
    return service({
        url: '/intervalTask/createIntervalTask',
        method: 'post',
        data
    })
}

export const stopIntervalTask = (params) => {
    return service({
        url: '/intervalTask/stopIntervalTask',
        method: 'get',
        params
    })
}
export const deleteIntervalTask = (params) => {
    return service({
        url: '/intervalTask/deleteIntervalTask',
        method: 'delete',
        params
    })
}
